src/rules/valid-mocked-module-path.ts

```diff
@@ -0,0 +1,99 @@
+import { statSync } from 'fs';
+import path from 'path';
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import {
+  createRule,
+  findModuleName,
+  getAccessorValue,
+  isSupportedAccessor,
+  isTypeOfJestFnCall,
+} from './utils';
+
+export default createRule({
+  name: __filename,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow mocking of non-existing module path',
+    },
+    messages: {
+      invalidMockModulePath: 'Mocked module path {{moduleName}} does not exist',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      CallExpression(node: TSESTree.CallExpression): void {
+        const { callee } = node;
+
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+
+        const { property } = callee;
+
+        if (
+          !node.arguments.length ||
+          !isTypeOfJestFnCall(node, context, ['jest']) ||
+          !(
+            isSupportedAccessor(property) &&
+            ['mock', 'doMock'].includes(getAccessorValue(property))
+          )
+        ) {
+          return;
+        }
+
+        const [nameNode] = node.arguments;
+        const moduleName = findModuleName(nameNode);
+
+        /* istanbul ignore if */
+        if (!moduleName) {
+          throw new Error(
+            'Cannot parse mocked module name from `jest.mock` -  - please file a github issue at https://github.com/jest-community/eslint-plugin-jest`',
+          );
+        }
+
+        try {
+          if (moduleName.value.startsWith('.')) {
+            const resolvedModulePath = path.resolve(
+              path.dirname(context.filename),
+              moduleName.value,
+            );
+
+            const hasPossiblyModulePaths = ['', '.js', '.ts']
+              .map(ext => `${resolvedModulePath}${ext}`)
+              .some(modPath => {
+                try {
+                  statSync(modPath);
+
+                  return true;
+                } catch {
+                  return false;
+                }
+              });
+
+            if (!hasPossiblyModulePaths) {
+              throw { code: 'MODULE_NOT_FOUND' };
+            }
+          } else {
+            require.resolve(moduleName.value);
```

Comment:
we can do an early return to avoid this `else`, and if you invert the condition we can deindent the other (larger) code path